fontstash/fontstash.h

```diff
@@ -1567,7 +1615,13 @@ static __inline void fons__vertices(FONScontext* stash, FONSquad q, FONSstate* s
     fons__vertex(stash, q.x1, q.y1, q.s1, q.t1, state->color);
 }
 
-void fonsSetShaping(FONScontext* stash, const char* script, const char* direction, const char* language)
+void fonsInitShapingPlan(FONScontext* stash, const char* str) {
+    FONSUTF32string fons32String = stash->params.toUTF32Alloc(stash->params.userPtr, str);
+    fons__hb_initShapingPlan(stash, fons32String);
+    stash->params.UTF32Free(stash->params.userPtr, fons32String);
+}
```

Comment:
Got to read the code. glyph_infos is available directly after buffer_add (Thats what I was originally looking for). Right above buffer_add implementation is 
https://github.com/behdad/harfbuzz/blob/master/src/hb-buffer.cc#L660

buffer_add sets the 'info' that is returned by get_glyph_infos:
https://github.com/behdad/harfbuzz/blob/master/src/hb-buffer.cc#L1516
https://github.com/behdad/harfbuzz/blob/master/src/hb-buffer.cc#L268